apps/daimo-mobile/src/view/screen/SendScreen.tsx

```diff
@@ -214,6 +254,140 @@ function SendPayment({ recipient }: { recipient: Recipient }) {
   );
 }
 
+function CreateNote({ hide }: { hide: () => void }) {
+  const [ephemeralPrivateKey, setEphemeralPrivateKey] =
+    useState<`0x${string}`>("0x");
+
+  const { chain } = useContext(ChainContext);
+  assert(chain != null);
+
+  const [dollars, setDollars] = useState(0);
+
+  const [account] = useAccount();
+  assert(account != null);
+
+  const [isNotesContractApproved, setIsNotesContractApproved] = useState(false);
+
+  useEffect(() => {
+    (async () => {
+      const allowance = await fetchNotesContractAllowance(
+        chain.clientL2,
+        account.address
+      );
+      setIsNotesContractApproved(allowance > 0);
+
+      const privKey = generatePrivateKey();
+      setEphemeralPrivateKey(privKey);
+    })();
+  }, []);
+
+  const createTxHandle = (
+    ephemeralPrivateKey: `0x${string}`,
+    isNotesContractApproved: boolean,
+    dollars: number
+  ): TxHandle => {
+    return async (account: DaimoAccount) => {
+      const ephemeralOwner = privateKeyToAccount(ephemeralPrivateKey).address;
+      if (isNotesContractApproved) {
+        console.log(
+          `[ACTION] skipping approve, creating note worth $${dollars}`
+        );
+        return account.createEphemeralNote(ephemeralOwner, `${dollars}`);
+      } else {
+        console.log(
+          `[ACTION] approving notes contract to spend token and creating note worth $${dollars}`
+        );
+        return account.erc20approveAndCreateEphemeralNote(
```

Comment:
looks excellent

in future, we might pre-approve both the notes contract & the paymaster in the account constructor